### PR TITLE
Added problem groups administration

### PR DESCRIPTION
**Pull request by gogo4ds**
Add Administration for managing Problem groups
Closes https://github.com/SoftUni-Internal/suls-issues/issues/3780

### CODE REVIEW

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemGroupsController.cs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+            if (result.IsError)
+            {
+                this.ModelState.AddModelError(string.Empty, result.Error);
+            }
+
+            return this.GridOperation(request, model);
+        }
+
+        public ActionResult Contest(int? id)
```
Comment (jicata) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemGroupsController.cs]: Better naming. Consider using verb-orientated naming instead.

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemGroupsController.cs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                    new { area = GlobalConstants.AdministrationAreaName });
+            }
+
+            this.ViewBag.ContestId = id;
+
+            return this.View(GlobalConstants.Index);
+        }
+
+        [HttpPost]
+        public JsonResult ProblemsInProblemGroup([DataSourceRequest]DataSourceRequest request, int id)
```
Comment (jicata) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemGroupsController.cs]: Better naming. Consider using verb-orientated naming instead.

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemGroupsController.cs (hunk 1 above)
Comment (gogo4ds) [reply Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemGroupsController.cs]: I think that the URL would be more user friendly if the name is just Contest: /ProblemGroups/Contest/325

Comment (jicata) [reply Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemGroupsController.cs]: You can consider using [attribute routing in this case](https://blogs.msdn.microsoft.com/webdev/2013/10/17/attribute-routing-in-asp-net-mvc-5/)

**Pull request merged**